server/protocols/jmap/src/main/java/org/apache/james/jmap/mailet/filter/ContentMatcher.java

```diff
@@ -57,6 +61,26 @@ private AddressHeader(String fullAddress) {
                 return Optional.empty();
             }
         }
+
+        boolean exactMatch(AddressHeader other) {
+            return fullAddress.equalsIgnoreCase(other.fullAddress)
+                || OptionalUtils.matches(address, other.address, String::equalsIgnoreCase)
+                || OptionalUtils.matches(personal, other.personal, String::equalsIgnoreCase);
```

Comment:
should be `fullAddress || ( address && personal )`, no?
else you match `"me" <me@example.com>` with `"me" <notme@example.com>`?

Reply:
What you describe should be true, but only when the address or personal optionals are not empty...

Reply:
Well, these tests were broken.

TDD lead me to complexify the matching condition structure.